Source/Tools/FEXLoader/LinuxSyscalls/GdbServer.cpp

```diff
@@ -1312,13 +1416,21 @@ void GdbServer::OpenListenSocket() {
   strncpy(addr.sun_path, GdbUnixSocketPath.data(), sizeof(addr.sun_path));
   size_t SizeOfAddr = offsetof(sockaddr_un, sun_path) + GdbUnixSocketPath.size();
 
+  auto TryBind = [](int Socket, struct sockaddr_un *addr, size_t Size) -> bool {
+    int Result = bind(Socket, reinterpret_cast<struct sockaddr*>(addr), Size);
+    return Result == 0;
+  };
+
   // Bind the socket to the path
-  int Result = bind(ListenSocket, reinterpret_cast<struct sockaddr*>(&addr), SizeOfAddr);
-  if (Result == -1) {
-    LogMan::Msg::EFmt("[GdbServer] Couldn't bind AF_UNIX socket '{}': {} {}\n", addr.sun_path, errno, strerror(errno));
-    close(ListenSocket);
-    ListenSocket = -1;
-    return;
+  if (!TryBind(ListenSocket, &addr, SizeOfAddr)) {
```

Comment:
Using a lambda here for just one repetition adds more noise than it really saves. Maybe a for loop does the trick more easily?

```cpp
for (int attempt = 0; attempt < 2; ++attempt) {
    ...
}
```